oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -1393,7 +1393,7 @@ private void updateCoords(String cluster, BundleJob bundle,
 
             //calculate next start time based on delay.
             endTime = (delay == null) ? endTime
-                    : EntityUtil.getNextStartTime(coord.getStartTime(), delay, EntityUtil.getTimeZone(entity), endTime);
+                    : EntityUtil.getNextInstanceTimeWithDelay(endTime, delay, EntityUtil.getTimeZone(entity));
```

Comment:
EffectiveTime is being sent for update. EndTime is sent only in case of a bundle update where it is supposed to be end time as the one mentioned in the entity cluster's validity.